system/lib/libc/musl/src/internal/pthread_impl.h

```diff
@@ -150,4 +171,11 @@ void __restore_sigs(void *);
 #define DEFAULT_STACK_SIZE 81920
 #define DEFAULT_GUARD_SIZE PAGE_SIZE
 
+#define __ATTRP_C11_THREAD ((void*)(uintptr_t)-1)
+
+#if !__EMSCRIPTEN_PTHREADS__
+void __emscripten_init_pthread_stub(void);
```

Comment:
Because the `pthread_t` structure for main thread is now unconditionally needed in all builds, I think we want to have a machinery that initializes the pthread control field in main thread identical to whether we are building with threads or not. Is it possible to reuse the same code and avoid these `pthread_stub` fields?